components/parameter/parameter.go

```diff
@@ -0,0 +1,491 @@
+	Default           interface{}         `json:"default,omitempty"`
+	Min               int64               `json:"minimum,omitempty"`
+	Max               int64               `json:"maximum,omitempty"`
+	MinLen            int64               `json:"minLength,omitempty"`
+	MaxLen            int64               `json:"maxLength,omitempty"`
+	Pattern           string              `json:"pattern,omitempty"`
+	MaxItems          int64               `json:"maxItems,omitempty"`
+	MinItems          int64               `json:"minItems,omitempty"`
+	UniqueItems       bool                `json:"uniqueItems,omitempty"`
+	MultipleOf        int64               `json:"multipleOf,omitempty"`
+	CollenctionFormat string              `json:"collectionFormat,omitempty"`
+}
+
+// https://swagger.io/specification/v2/#schema-object
+type JsonResponseSchema struct {
+	Ref   string                   `json:"$ref,omitempty"`
+	Type  string                   `json:"type,omitempty"`
+	Items *JsonResponseSchemeItems `json:"items,omitempty"`
+}
+
+type JsonResponseSchemeItems struct {
+	Type string `json:"type,omitempty"`
+	Ref  string `json:"$ref,omitempty"`
+}
+
+// Parameter represents a parameter in an API endpoint.
+type Parameter struct {
+	name             string
+	typeValue        ParamType
+	in               Location
+	required         bool
+	description      string
+	enum             []interface{}
+	defaultValue     interface{}
+	format           string
+	min              int64
+	max              int64
+	minLen           int64
+	maxLen           int64
+	pattern          string
+	maxItems         int64
+	minItems         int64
+	uniqueItems      bool
+	multipleOf       int64
+	collectionFormat CollectionFormat
+}
+
+// GetLocation returns the location of the parameter (i.e. Query, Body, Path, and etc.)
+func (p Parameter) GetLocation() Location {
+	return p.in
+}
+
+// AsJson returns the json representation of Parameter
+func (p *Parameter) AsJson() JsonParameter {
+	return JsonParameter{
+		Name:              p.name,
+		In:                p.in.String(),
+		Description:       p.description,
+		Required:          p.required,
+		Type:              p.typeValue.String(),
+		Format:            p.format,
+		Enum:              p.enum,
+		Default:           p.defaultValue,
+		Min:               p.min,
+		Max:               p.max,
+		MinLen:            p.minLen,
+		MaxLen:            p.maxLen,
+		Pattern:           p.pattern,
+		MaxItems:          p.maxItems,
+		MinItems:          p.minItems,
+		UniqueItems:       p.uniqueItems,
+		MultipleOf:        p.multipleOf,
+		CollenctionFormat: p.collectionFormat.String(),
+	}
+}
+
+// NoParam is an empty slice of parameters.
+var NoParam []Parameter
+
+// validate will check and correct certian parameter configurations that cause non-fatal OpenAPI warning/errors when rendering.
+// e.g. Options being set for a parameters `In == Path`, while `required == false` is a non-fatal syntactical error.
+func (p *Parameter) validate(l Location) {
```

Comment:
Yup, I started to then I realized that to do this right it would be a bigger effort than i was ready to do now. Forgot to remove after quitting on that idea. It's gone now.